src/Orleans/Statistics/ThreadCycleStopWatch.cs

```diff
@@ -98,10 +98,53 @@ private static class NativeMethods
         {
             [DllImport("Kernel32.dll")]
             public static extern bool QueryThreadCycleTime(IntPtr handle, out ulong cycles);
+
             [DllImport("Kernel32.dll")]
             public static extern bool QueryPerformanceFrequency(out long lpFrequency);
+
             [DllImport("Kernel32.dll")]
-            public static extern IntPtr GetCurrentThread();    
+            public static extern IntPtr GetCurrentThread();
+        }
+
+        #region IDisposable Support
+
+        private bool disposedValue = false; // To detect redundant calls
+
+        protected virtual void Dispose(bool disposing)
```

Comment:
I wonder if we even need to do anything here? The class doesn't actually hold any native resources that it needs to release. Maybe just suppress the warning with a comment about that instead of adding effective no-op code?